openedx/core/djangoapps/credit/tests/test_api.py

```diff
@@ -0,0 +1,160 @@
+""" Tests for credit course api """
+
+from opaque_keys.edx.keys import CourseKey
+from openedx.core.djangoapps.credit.api import (
+    get_credit_requirements, set_credit_requirements, _get_requirements_to_disabled
+)
+from openedx.core.djangoapps.credit.exceptions import InvalidCreditRequirements
+from openedx.core.djangoapps.credit.models import CreditCourse, CreditRequirement
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+
+
+class ApiTestCases(ModuleStoreTestCase):
+    """ Tests for credit course api """
+
+    def setUp(self, **kwargs):
+        super(ApiTestCases, self).setUp()
+        self.course_key = CourseKey.from_string("edX/DemoX/Demo_Course")
+
+    # def test_set_credit_requirements_invalid_credit_course(self):
+    #     with self.assertRaises(InvalidCreditRequirements):
+    #         requirements = [
+    #             {
+    #                 "namespace": "grade",
+    #                 "name": "grade",
+    #                 "configuration": {
+    #                     "min_grade": 0.8
+    #                 }
+    #             }
+    #         ]
+    #         set_credit_requirements(self.course_key, requirements)
```

Comment:
added new kind of exception "InvalidCreditCourse" if the given course is not marked as the credit